flask_session/sessions.py

```diff
@@ -485,11 +489,10 @@ def open_session(self, app, request):
             saved_session = None
         if saved_session:
             try:
-                val = saved_session.data
-                data = self.serializer.loads(str(val))
+                data = self.serializer.loads(saved_session.data)
                 return self.session_class(data, sid=sid)
             except:
-                return self.session_class(sid=sid)
+                raise
```

Comment:
Development Artifact. In earlier iterations I raised a custom keg error code, good catch.